### DISCUSSION TITLE
Update to documentation.

### DISCUSSION THREAD

**Discussion by OdifYltsaeb** [avitorio/outstatic#259] [Ideas]
I'm trying to get my first project outstatic/nextjs off the ground and I'm running headfirst into missing documentation issues. Without whining, I'll outline them here:
- The first startup page image (this one: https://outstatic.com/docs/images/outstatic-welcome-U1ND.png) says that OST_REPO_SLUG is missing, yet it does not explain what should be the value for it. Also you mention that OST_REPO_OWNER is "optional" which makes me think that that the slug should invude username (OdifYltsaeb in my case) . Yet when you set it to OdifYltsaeb/repo_name and set OST_REPO_OWNER to OdifYltsaeb, then the application will complain, that the repo slug should not contain the username. **TODO**: Please provide a clear example of what is the repo slug. Clarify right away, that username should not be a part of the repo slug.
- At no point is it clarified what kind of repo it should be. Should it be a new and empty repo just for content? Should it be the same repo where the rest of the project will be? Can I keep the content separated from the rest of the application? 

I think I should put the following to issues/bugs instead. But maybe I'm missing something, and they are not bugs at all, but rather something I could fix/solve with better documentation. This is why I file this request under update to documentation.
- My project runs at 127.0.0.1:8000 locally. This is what my /outstatic page looks like:
![image](https://github.com/user-attachments/assets/6607144f-367a-4062-8398-168776530d37)
None of the requests are 404 so I'm not sure what is missing,  but you can see that it's broken. No idea why. The other pages work just fine.
- I'm note able to create a new collection. It fails with this error: Oops! We couldn't create your collection. Please, make sure your settings are correct by [clicking here](http://127.0.0.1:8000/outstatic/settings) .
- Once I go to settings page I am unable to actually change anything there. My main branch is "master" yet the settings page says this:
![image](https://github.com/user-attachments/assets/de10eabf-f9b3-42cc-82b6-564700def3be)

Other documentation that could be improved upon:
- Environment variables:
- `OST_CONTENT_PATH` - # Where content is saved. Defaults to outstatic/content if empty. - What is kind of path is this outstatic/content? Local folder? which is the parent? Where?
- `OST_MONOREPO_PATH` - # Specify monorepo folder of your Outstatic content. What do you mean monorepo folder? What is that path there? What is monorepo on your sense? Does it mean that If I have nextjs project with outstatic I now have a monorepo since it hosts 2 projects?
- 

In any case, I've not been able to get the project up and running, and I need help. Or please provide clearer and better examples in your documentation. If you manage to help me I'll come up with suggested documentation changes that would have prevented this post being ever posted.

**Comment by avitorio** [comment]
Hey @OdifYltsaeb,

Sorry to hear you're having trouble with your setup.
I've updated the documentation to give more explicit info on the env vars.

The screenshot you shared seems to be a problem with css not being loaded. Did you create your `layout.tsx` in the order and with the contents specified in the docs?

You shouldn't need the `OST_CONTENT_PATH` nor the `OST_MONOREPO_PATH` for your project. They are optional.

And lastly, what version of Next.js are you using? We currently don't support version 15 yet.

If you need more support, here's a link with some of the [support options](https://github.com/sponsors/avitorio?frequency=one-time&sponsor=avitorio). 

I hope this helps.

**Comment by OdifYltsaeb** [comment]
hey @avitorio , Thank you for responding!

- My nextjs version is 14.2.14,
- I created layout.tsx following this page/guide 1 to 1 https://outstatic.com/docs/getting-started#adding-outstatic-to-a-nextjs-website. Only difference is that I skipped the typescript and used js instead. so my page.tsx is page.jsx instead with contents l like
```
import 'outstatic/outstatic.css';
import { Outstatic } from 'outstatic';
import { OstClient } from 'outstatic/client';

export default async function Page({ params }) {
    const ostData = await Outstatic();
    return <OstClient ostData={ostData} params={params} />;
}

```
- as for the assumption of CSS not being loaded. No requests made by the site result in errors. It seems everything is being loaded just fine. There's something weird going on there. For example. If I use "new collection" link on the `/outstatic` page to navigate to collections view ( `outstatic/collections/new` ), then the next page loads nice and pretty. Now if I use the "Outstatic" logo in the top-left corner to navigate back to '/outstatic', then the same view, which was broken before is now nice and pretty. If I reload the page, though, then it will break again. Same with other views. If I reload the page on `/outstatic/collections/new' then the page is "broken" again.
 
![image](https://github.com/user-attachments/assets/529d7239-5cb8-4380-b2ce-eb35468241be)

**Reply by avitorio** [reply]
I really can't think of any reason why this is happening. 
The fact that other pages load fine and clicking the back button applies the styles is very odd.

Have you tried doing the Deploy with Vercel option we have on the website and then cloning the project locally?
Are you installing Outstatic on an existing website or starting from scratch?

**Reply by OdifYltsaeb** [reply]
I created a new nextjs project using my own template. There are no other pages, and it's rather simple at this point. Only a docker compose to get the project up and running. I made the project public, you can check it out yourself: https://github.com/OdifYltsaeb/olem_landing just run make build and make docker to get it up and running in linux environment.

As for vercel - no. I intend to stay away from it and similar services as far as possible and host the whole project myself in dockersied environment... or perhaps using pm2.

**Reply by avitorio** [reply]
Have you tried removing the app/layout.jsx file?

I suggested Vercel because then you'd have a working example you could build upon and deploy wherever you want later on.

**Reply by OdifYltsaeb** [reply]
I  have not. 

I did though, and on root page this happens (so it's not a permanent solution):
![image](https://github.com/user-attachments/assets/f2d4336d-f368-4213-974e-a118db7f8138)

but on `/outstatic` the page loads up just fine with no problem.

I did discover another error, though. I start authentication process at HTTP://127.0.0.1:8000/outstatic. It authenticates me, but redirects me back to HTTP://localhost:8000 instead not back to the URL I came from. Also this is the oauth application settings:
![image](https://github.com/user-attachments/assets/8985bd9d-a551-41d7-b173-df3444a4545d)
so there's no reason it should redirect me back to localhost.

**Reply by avitorio** [reply]
So, the problem with /outstatic not loading correctly is the fact your app/layout.jsx is interfering with it. You should move or combine your app/layout.jsx with the one inside your [(unauthenticated)](https://github.com/OdifYltsaeb/olem_landing/tree/master/app/src/app/(unauthenticated)) folder. 

As for returning to localhost, there are no mentions of localhost in the outstatic package. I can't really think of the reason for it to be redirecting to localhost:3000. Are you using the correct `OST_GITHUB_ID` and `OST_GITHUB_SECRET`? Have you tried logging in on a different browser or on a incognito window?

**Reply by OdifYltsaeb** [reply]
Hmhh Allright. I'll move the stuff in my layout elsewhere... if I really need it. That's a working solution I guess. It would be interesting to figure out why and how does it interfere with what you're doing, though. I did some commenting in/out in layout and it seems, that the page breaks, when I comment back in `<ToastProvider />` component. Since the component really does nothing but wrap ToastContainer from react-toastify I have no idea wth could be going on. I guess I can move everything to (unauthenticated)/layout.jsx, though. 

As for redirecting back to localhost, yes all the values are correct. Verified them against the oauth app. And Yes I did try everything in incognito window. I even removed the middleware.js ( which is a remnant from another project ) just to check if it affects anything - and it shouldn't, as none of the paths matches.

**Reply by OdifYltsaeb** [reply]
is this how it is supposed to work? 
- I make changes using the outstatic views. The data is then deployed to github in outstatic/content folder. I
- Locally I see no changes when using db.find. Only when I run git pull do the changes appear. populating metadata seems to be doing nothing. It seems there's still something wrong with the whole setup.

**Reply by avitorio** [reply]
Hey @OdifYltsaeb ,

Unfortunately I'm currently filled with work and won't have time to engage with helping you debug today. I'm also traveling tomorrow. I'll only be back online on Monday.

I suggest you look at our examples in the repository to see how things should be set up.

The source of truth for information on Outstatic is your GitHub repo, so you need to pull changes to see them locally.

I hope this helps.

**Reply by OdifYltsaeb** [reply]
@avitorio It's all good, man. I appreciate you trying to help at all. You've done something great and built a ready to use tool like this. I'ts insane effort. Believe me- I understand. I've not been able to do the same. So If you don't have time to deal with me it's fine. I already greatly appreciate your response.

I'm just trying to see if I can use outstatic as a simple tool for one basic project and to learn new tools. I'll go ahead and look at the codebase, but the codebase don't easily yield how you anticipate user flows. Hence the question about how you intend the work flows to be used. In any case - take care, man. Answer when you can and where you can. It's all good and much appreciated.

**Reply by OdifYltsaeb** [reply]
@avitorio, when you do get the chance, can you explain more about the workflow?

- Does it make sense to create a sub repository for the content or host the content along with the code of the website? But I guess that comes with all the other issues you get when you work with git submodules...
- Do I have to `git pull` after every change in the content for the content to be available through the `db.load(...` or is there something wrong with my setup if I have to? Because after saving the changes, I have to manually do a git pull for the db.load to yield the changed documents.

**Reply by avitorio** [reply]
Hey,
I don't think the submodules approach is a good one. You generally want your markdown files to be part of your project.

And yes, you need to git pull after each change.

**Reply by OdifYltsaeb** [reply]
Awesome. Thank you! Learning how the tool is supposed to work is very helpful for me. Think we can close this one.
